www-reputation-credentials-protocol.md

```diff
@@ -0,0 +1,265 @@
+# Mitigating Channel Jamming with One-Shot HTLC Forward Reputation Credentials
+
+Authors: Antoine Riard <btc@ariard.me>
+
+Created: 2022-11-15
+
+# Problem Statement
+
+The channel jamming issue has been one of the longuest standing and unsolved issue affecting the
+Lightning Network. Not only attack has been demonstrated in practice, but also theoritically
+[studied](https://jamming-dev.github.io/book/): "A channel jamming is when a malicious entity blocks up liquidity in the LN,
+by making a payment to themselves, via third party channels and then never revealing the secret, such
+that the payment never completes".
+
+The channel jamming root cause comes from the following situation. A routed HTLC across multiple Lightning
+hops is a chain of contracts among trust-minimized counterparties. The settlement of this chain of contracts
+is uniform in its outcome (success/failure) across all the intermediary counterparties, however such counterparties
+cannot influence the outcome, once the chain of contract is established. As setting up this chain of contracts
+consume scarce resources (i.e the channel liquidity timevalue), there is a counterparty risk without guarantee
+there is an adequate compensation. Indeed, the routing fees are only paid in case of success.
+
+# Solution introduction
+
+This counterparty risk can be solved straightforwardly by the introduction of a premimum fee carried on by
+the HTLC sender. This jamming solution known as ["upfront fees"](https://github.com/t-bast/lightning-docs/blob/master/spam-prevention.md)
+in the Lightning community, aims to a fair distribution of the HTLC forward risk among all the chain of
+contracts counterparties. While there is a concern this risk distribution doesn't capture well the
+shenanigans of a HTLC forward (e.g offline routing hop, spontaneously congestioned channel), there is a
+major drawback as it introduces a permanent overhead fee for the network participants.
+
+In contract, another type of reputation-based solution can be introduced by formalizing a new assumption
+"Reputation incentives HTLC senders to behave-well, if there is an adequate proportion between the
+acquisition cost and the amount of resources credited in function of said reputation". This assumption is
+in fact analoguous to the one underpining routing scoring algorithms incentiving the routing hops reliability.
+A reputation system attached to the HTLC forward, not only can protect against malicious jamming entity but
+also can offer the traffic shapping tooling for node operators to improve the quality of their inbound HTLC
+across time.
+
+# Proposed solution
```

Comment:
I think it may be useful too - if only as a thought experiment - to envision the simplest possible version of the scheme. So maybe just a single credential per htlc, no introduction of liquidity units and avoiding the use of onion messages?

Reply:
I think this is a good point, to work out the simplest possible version of the scheme.

A single credential per HTLC with a fixed cost-to-liquidity (10 sat of pre-paid for 1 BTC of liquidity for 2016 blocks). I wonder if in fact you really need liquidity unit if we can have global agreement on the credential granularity-level.

Reply:
This would make up a good section for this document :)